File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/oracle/NumericTypes.scala
package dev.typr.foundations.docs.oracle
import dev.typr.scalafoundations.*
import dev.typr.scalafoundations.data.*

@SuppressWarnings(Array("unused"))
object NumericTypes:
  //start
  val numberType: OracleType[BigDecimal] = OracleTypes.number
  val decimal: OracleType[BigDecimal] = OracleTypes.number(10, 2)  // NUMBER(10,2)
  val intType: OracleType[Int] = OracleTypes.numberAsInt(9)    // NUMBER(9)
  val longType: OracleType[Long] = OracleTypes.numberAsLong(18)  // NUMBER(18)
  //stop

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundations/docs/postgresql/NumericTypes.kt
package dev.typr.foundations.docs.postgresql

import dev.typr.kotlinfoundations.*
import dev.typr.kotlinfoundations.data.*
import dev.typr.foundations.data.Money
import java.math.BigDecimal

@Suppress("unused")
class NumericTypes {
    //start
    val intType: PgType<Int> = PgTypes.int4
    val decimalType: PgType<BigDecimal> = PgTypes.numeric
    val moneyType: PgType<Money> = PgTypes.money
    //stop
}
